### PR TITLE
fixing eventloop blockage due to sync methods

### PR DESCRIPTION
**Pull request by assafchamoy**

### CODE REVIEW

**Comment by assafchamoy** [comment]
still not working, 
Should work more on crypto utils.

**Comment by assafchamoy** [comment]
Fixed performance issues:
- Transformed all operations with files to asynchronous. 
- Changed unique id's generation algorithm. 
- Filter unnecessary files before iterations.

File: src/common/utilities/dirOperations.ts [hunk 1]
```diff
@@ -41,47 +43,57 @@ class DirOperations {
...
     this.logger.info(`[DirOperations][getDirectoryContent] fetching directory of path ${path as string}`);
-    const isDirExists = existsSync(path);
+    const isDirExists = await this.checkFileExists(path);
 
     if (!isDirExists) {
       throw new NotFoundError(StorageExplorerErrors.FILE_NOT_FOUND);
     }
 
-    const isDir = statSync(path).isDirectory();
+    const isDir = await statPromise(path);
```
Comment (assafchamoy) [inline src/common/utilities/dirOperations.ts:77]: change var name

**Review by alebinson — APPROVED** [review]
FANTASTICO!!!

**Pull request merged**